Assets/Scripts/Town/PlayerInteractions.cs

```diff
@@ -20,27 +22,48 @@ void Update()
                     useLayerMask = true,
                     useTriggers = true
                 },
-                result);
+                newInteractables);
 
-            var newInteractables = new HashSet<IInteractable>(result.Select(x => x.GetComponent<IInteractable>()));
+            float closestInteractibleDistance = float.PositiveInfinity;
+            Collider2D closestInteractable = null;
 
             foreach (var interactable in newInteractables)
             {
-                if (!_oldInteractables.Contains(interactable))
+                // Maybe later change so you can choose which collider is used.
+                var playerInteractionCollider = GetComponent<Collider2D>();
```

Comment:
Można zapisać tą wartość jako pole w klasie, aktualnie w każdej kilka razy pobieramy collider gracza.

Nie jestem pewien czy na obiekcie interactable możemy zrobić interactable.transform.position? Jak tak, to nawet nie trzeba cashować collidera, tylko użyc _Vector2.Distance(a,b)_ lub _(a-b).magnitude_, gdzie **a** to transform.position, a **b** to interactable.transform.position.

Reply:
Użyłem Physics2D.Distance bo oblicza on najmniejszą odległość między colliderami. Gdybym użył transform.position to w zależności od położenia collidera względem transform.position w obiektach odległość interakcji mogła by się wydawać inna.